Update domdec.rst (#2322)
arbor-sim/arbor: doc/concepts/domdec.rst

.. glossary::

   cell_group

>       List of same-kind cells that share some information. Must not be split

Reviewer: ```suggestion
      Collection of cells with the same kind. Will be simulated as a unit. Must not be split
```

      across domains.

   domain

>       Produced by a `load_balancer`; a list of all `cell_groups`

Reviewer: ```suggestion
      Produced by a ``load_balancer``; a list of all ``cell_groups``
```

>       located on the same hardware. A `communicator` deals with the full set of

Reviewer: ```suggestion
      located on the same hardware. A ``communicator`` deals with the full set of
```

>       cells of one `domain`.

Reviewer: ```suggestion
      cells of one ``domain``.
```
